os/net/mac/framer/frame802154.c

```diff
@@ -294,6 +301,13 @@ field_len(frame802154_t *p, field_length_t *flen)
   /* init flen to zeros */
   memset(flen, 0, sizeof(field_length_t));
 
+  if((p->fcf.frame_type == FRAME802154_MPFRAME)
+      && (!p->fcf.long_frame_control)) {
+    flen->frame_ctrl_len = 1;
```

Comment:
I've been shrinking things for MSP430 lately, `x = cond ? 1 : 2;` will generate smaller code than assigning to the same variable in two if/else branches.